src/layer/vulkan/shader/pooling_adaptive.comp

```diff
@@ -72,28 +72,34 @@ void main()
 
     afp res;
 
+    // calculate adaptive kernel size
+    int sx = int(floor(afp(gx * psc(w)) / afp(psc(outw))));
+    int ex = int(ceil(afp((gx + 1) * psc(w)) / afp(psc(outw))));
+    int kernel_w = ex - sx;
+    int sy = int(floor(afp(gy * psc(h)) / afp(psc(outh))));
+    int ey = int(ceil(afp((gy + 1) * psc(h)) / afp(psc(outh))));
+    int kernel_h = ey - sy;
+
```

Comment:
What about resolving the kernel size on cpu side, and passing them via push constants ?

Reply:
But according to the implementation of adaptive pooling in c++ and pytorch, the kernel size is dynamic and calculated on every location. 

For example, input size and output size is 5 and 3, the kernel size for each data will be 2, 3, 2. 

Can this be done on cpu side(pooling_vulkan.cpp) since kernel size is no long constant?

ncnn c++:
https://github.com/Tencent/ncnn/blob/ab56083ca543d4315bef9913b23b843e1fdd8c76/src/layer/pooling.cpp#L114-L128

pytorch:
https://github.com/pytorch/pytorch/blob/51861cc9b19d9c483598e39932661822a826d3a2/aten/src/ATen/native/AdaptiveAveragePooling.cpp#L34-L52

Reply:
okay !